### PR TITLE
Refactor backend (part3)

### PR DESCRIPTION
**Pull request by niwinz**
This PR includes the migration from JETTY to UNDERTOW as http backend. This all is donde throught the library we are using: yetti.

1. It makes the http layer fully async (no need to sync operations on websocket handlers, limitation that can't be easy solved in jetty). And enables more control on how to model the execution model.
2. The session store now can be asynchronous (previously forced to by sync because of websockets); now we can think in implement it using redis.
3. Start using ring-2.0 upcoming changes that simplifies the request and response models and make them more performant.
4. The threading model change also improves the resource usages. Now we use probably half of threads for the same workload.

It also fixes some other, bugs already present on the codebase or introduced by part 2 refactor merge.

### CODE REVIEW

File: backend/src/app/http.clj [hunk 1]
```diff
@@ -10,99 +10,97 @@
...
 (s/def ::router some?)
 (s/def ::port ::us/integer)
 (s/def ::host ::us/string)
 (s/def ::name ::us/string)
-(s/def ::max-threads ::cf/http-server-max-threads)
-(s/def ::min-threads ::cf/http-server-min-threads)
+(s/def ::executors (s/map-of keyword? ::wrk/executor))
+
+;; (s/def ::max-threads ::cf/http-server-max-threads)
+;; (s/def ::min-threads ::cf/http-server-min-threads)
```
Comment (niwinz) [inline backend/src/app/http.clj:40]: I will improve and delete this in other MR, before the next release.

File: backend/src/app/metrics.clj [hunk 2]
```diff
@@ -265,9 +263,9 @@
...
-(defn instrument-jetty!
-  [^CollectorRegistry registry ^StatisticsHandler handler]
-  (doto (JettyStatisticsCollector. handler)
-    (.register registry))
-  nil)
+;; (defn instrument-jetty!
+;;   [^CollectorRegistry registry ^StatisticsHandler handler]
+;;   (doto (JettyStatisticsCollector. handler)
+;;     (.register registry))
+;;   nil)
```
Comment (hirunatan) [inline backend/src/app/metrics.clj:270]: May this be deleted?

Comment (niwinz) [reply backend/src/app/metrics.clj:270]: yes, i will do it on my next MR

File: backend/src/app/rpc/mutations/profile.clj [hunk 3]
```diff
@@ -407,43 +407,32 @@
...
-      ;; Schedule deletion of old photo
-      (when-let [id (:photo-id profile)]
-        (sto/touch-object! storage id))
-
-      ;; Save new photo
-      (db/update! pool :profile
-                  {:photo-id (:id photo)}
-                  {:id profile-id})
-      nil)))
+  [{:keys [pool storage executors] :as cfg} {:keys [profile-id] :as params}]
```
Comment (hirunatan) [inline backend/src/app/rpc/mutations/profile.clj:422]: Is the `info` command not needed now?

Comment (hirunatan) [reply backend/src/app/rpc/mutations/profile.clj:422]: Ok. I've seen you've moved it to upload-photo. False alarm.

File: backend/src/app/storage.clj [hunk 4]
```diff
@@ -257,7 +257,9 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
 ;; A task responsible to permanently delete already marked as deleted
-;; storage files.
+;; storage files. The storage objects are practically never marked to
+;; be deleted directly by the api call. The touched-gc is responsible
+;; collect the usage of the object and mark it as deleted.
```
Comment (hirunatan) [inline backend/src/app/storage.clj:262]: collect -> of collecting

File: backend/src/app/metrics.clj (hunk 2 above)
Comment (hirunatan) [reply backend/src/app/metrics.clj:270]: ok

**Pull request merged**